### ISSUE TITLE
Multiple OpenGL versions

### ISSUE THREAD

**Issue by EtienneBruines** [EngoEngine/engo#259]
Regarding our OpenGL version discussion we had yesterday. We might decide to support multiple OpenGL versions. As I've read, most of the "newer" functions available in higher versions, are usually available through "extensions" (you check if they are supported, and if so, you use them) - but we also might support multiple versions, and let the game-maker decide which one he wants to use. Since we don't have a multitude of shaders (yet), this is fairly easy to do. 

One of those newer functions is the VAO (Vertex Array Object), which basically allows us to define all Gl.VertexAttribPointer beforehand, and squash them into one OpenGL call (Currently we have to call 3 https://github.com/EngoEngine/engo/blob/master/shaders.go#L131-L133 per buffer-change (so basically per object)). This is probably something we can do while changing to core/engo (#245). 

We'd only have to figure out if we want a separate package per OpenGL version (so just the RenderSystem), or if we want to import both (not sure if that won't collide?), and simply ask the user to define the OpenGL version in the RunOpts (this is required anyway; but we might be able to use just 1 package, and use that setting to figure out which to use). 

Using two packages results in the smaller binary for sure. Using one package would probably let OpenGL2 and OpenGL3 calls to get put into the binary. Not that much of a difference (20MB or so? Maybe?) - but still worth noticing.

If we are to do this, it'd be nice to do for `1.0`. 

Please share (all of) your thoughts :smile:

**Comment by EtienneBruines** [comment]
> Dimitar Dimitrov @Newbrict 15:05
> I think we should wait on supporting multiple until it becomes an issue or is at risk of becoming one, we should be careful for feature creep
> 
> Harrison Shoebridge @paked 15:06
> ^
> 
> Etienne Bruines @EtienneBruines 15:06
> Feature creeping is indeed something to watch out for; I just want to cover most of the stuff that might involve a lot of refactoring. Whether we actually do it or not, is still up for discussion - but I'd rather discuss breaking stuff now, than after we say "this is 1.0"
> 
> Dimitar Dimitrov @Newbrict 15:07
> Good point, I didn't think it would be breaking
> 
> Harrison Shoebridge @paked 15:07
> I don't think it is?
> What if we had them implemented as different RenderSystems or something?

To summarize: this issue is on hold.
